File: software/examples/go/example_configuration.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/performance_dc_bricklet"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Performance DC Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	pdc, _ := performance_dc_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	pdc.SetDriveMode(performance_dc_bricklet.DriveModeDriveCoast)
	pdc.SetPWMFrequency(10000) // Use PWM frequency of 10 kHz
	pdc.SetMotion(4096, 4096)  // Slow ac-/deceleration (12.5 %/s)
	pdc.SetVelocity(32767)     // Full speed forward (100 %)
	pdc.SetEnabled(true)       // Enable motor power

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	// Stop motor before disabling motor power
	pdc.SetMotion(4096, 16384)          // Fast decceleration (50 %/s) for stopping
	pdc.SetVelocity(0)                  // Request motor stop
	time.Sleep(2000 * time.Millisecond) // Wait for motor to actually stop: velocity (100 %) / decceleration (50 %/s) = 2 s
	pdc.SetEnabled(false)               // Disable motor power
}

File: software/examples/uc/example_configuration.c
// This example is not self-contained.
// It requires usage of the example driver specific to your platform.
// See the HAL documentation.

#include "src/bindings/hal_common.h"
#include "src/bindings/bricklet_performance_dc.h"

void check(int rc, const char *msg);
void example_setup(TF_HAL *hal);
void example_loop(TF_HAL *hal);

static TF_PerformanceDC pdc;

void example_setup(TF_HAL *hal) {
	// Create device object
	check(tf_performance_dc_create(&pdc, NULL, hal), "create device object");

	check(tf_performance_dc_set_drive_mode(&pdc,
	                                       TF_PERFORMANCE_DC_DRIVE_MODE_DRIVE_COAST), "call set_drive_mode");
	check(tf_performance_dc_set_pwm_frequency(&pdc,
	                                          10000), "call set_pwm_frequency"); // Use PWM frequency of 10 kHz
	check(tf_performance_dc_set_motion(&pdc, 4096,
	                                   4096), "call set_motion"); // Slow ac-/deceleration (12.5 %/s)
	check(tf_performance_dc_set_velocity(&pdc,
	                                     32767), "call set_velocity"); // Full speed forward (100 %)
	check(tf_performance_dc_set_enabled(&pdc,
	                                    true), "call set_enabled"); // Enable motor power
}

void example_loop(TF_HAL *hal) {
	// Poll for callbacks
	tf_hal_callback_tick(hal, 0);
}
